src/jdk.internal.le/share/classes/jdk/internal/org/jline/JdkConsoleProviderImpl.java

```diff
@@ -83,6 +83,30 @@ public Reader reader() {
             return terminal.reader();
         }
 
+        @Override
+        public JdkConsole println(Object obj) {
+            writer().println(obj);
```

Comment:
the result of 'writer()' can be stored in a local variable (printing code are not JITed as often as the rest of the codes)

Reply:
I assume it's about performance. If so, I would defer any performance-related tweaks until they are necessary. Interactive reading from Console does not sound like something requiring that level of performance tweaking.

Reply:
yes, let see what @cl4es will say when he will test the time to print "hello world"

Reply:
@forax I don't think anything touched on here is used during bootstrap; perhaps there are apps we could cover that uses these APIs, but I think for line-based console IO a few calls in the interpreter is not going to make a noticeable difference.